Add a runs list

lukebjerring opened the pull request:
Similar prototype to https://github.com/w3c/wptdashboard/compare/sha-overlay, but doesn't pull from git, just uses what exists locally.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/wptdashboard/188)
<!-- Reviewable:end -->

lukebjerring commented:
Makes progress towards https://github.com/w3c/wptdashboard/issues/172

lukebjerring commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[test_run_handler.go, line 109 at r2](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPc-r2-109:-Kxx6GJHpbc3d3Q4UTkP:b-896fix) ([raw file](https://github.com/w3c/wptdashboard/blob/79af9f98c12fc00dce9f64dd9ac5d44295865a93/test_run_handler.go#L109)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

There's an extra newline here
</blockquote></details>

Done.

---

*[components/wpt-runs.html, line 40 at r2](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPZ-r2-40:-Kxx5IH_oaTD6-VmUbf_:b-896fix) ([raw file](https://github.com/w3c/wptdashboard/blob/79af9f98c12fc00dce9f64dd9ac5d44295865a93/components/wpt-runs.html#L40)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

There's some extra unused CSS here.
</blockquote></details>

Done.

---

*[components/wpt-runs.html, line 154 at r2](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPZ-r2-154:-Kxx68q3_gBMSusWrcUo:b-ild2yz) ([raw file](https://github.com/w3c/wptdashboard/blob/79af9f98c12fc00dce9f64dd9ac5d44295865a93/components/wpt-runs.html#L154)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

The indentation for JS in this project should be 2 spaces (unfortunately it looks like we don't have a lint rule for that yet ¯\\_(ツ)_/¯)
</blockquote></details>

NP - Changed my gogland preferences for the project.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wptdashboard/188)*
<!-- Sent from Reviewable.io -->

jeffcarp commented:
I tested this out locally and it looks good. A few things:

- We can't display total passing percentages for browsers (being anti-gamification is one of the goals of the project)
- It'd be great to have a link in the header to this page
- Can you add an explanation at the top saying "these are the last 10 runs per browser"?

lukebjerring commented:
- Removed summaries ( :'( that was the slowest/hardest thing to write in the PR )
    - Added 'present' or 'missing' classes, which currently just set the bg-color (let's something cooler later.)
- Added link
- Added info bar (temporarily. I'd like to follow this up with some basic pagination query params)

---

Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-:-Ky1Wbv_JMRcoZNt7b7Z:b-pfsk3e)*
<!-- Sent from Reviewable.io -->

jeffcarp commented:
I deployed this branch here:

https://20171103t154422-dot-wptdashboard.appspot.com/

Looks great!! :smile_cat: 

> I'd like to follow this up with some basic pagination query params

You can increase the query to 100 for now which will alleviate some need for pagination.

Also I will add a section to the docs on making sure we don't surface "scores" or overall percentages in the UI. (So we have an easy place to point to)

Will look this over another time when I get a sec and deploy if everything looks good.

jeffcarp commented on components/wpt-runs.html:
```diff
@@ -0,0 +1,146 @@
+      <template is="dom-repeat" items="{{ testResults }}" as="results">
+        <tr>
+          <td>
+            <a href="/?run={{ results.sha }}">{{ results.sha }}</a>
```
This needs to be updated to be `sha=`

jeffcarp commented on components/wpt-runs.html:
```diff
@@ -0,0 +1,146 @@
+        background-color: #e2e3fd;
+      }
+      .present {
+        background-color: #e3fde2;
```
These colors are a little confusing since green rectangles in the dashboard so far have only meant "tests passing." How about these:

```
.missing {
  background-color: #e3edfd;
}
.present {
  background-color: #337af3;
}
```

![screenshot from 2017-11-06 12 08 46](https://user-images.githubusercontent.com/916028/32461616-4c00161a-c2eb-11e7-92f7-51f08b101d92.png)

jeffcarp commented on components/wpt-runs.html:
```diff
@@ -0,0 +1,146 @@
+      tr td {
+        padding: 0 0.5em;
+        margin: 2px;
+      }
```
Can you add:

```css
a {
  text-decoration: none;
  color: #0d5de6;
  font-family: monospace;
}
```

(it's from the `<wpt-results>` component - I don't know a better way of sharing CSS between components :crying_cat_face:)

lukebjerring replied:
Done.

lukebjerring commented:
Review status: 0 of 7 files reviewed at latest revision, 6 unresolved discussions.

---

*[components/wpt-runs.html, line 30 at r5](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPZ-r5-30:-KyHzbxFlo2ob9wfOiNZ:b-896fix) ([raw file](https://github.com/w3c/wptdashboard/blob/71920c92ce6f20feb4ecefbefc1056d98a622b11/components/wpt-runs.html#L30)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

Can you add:

```css
a {
  text-decoration: none;
  color: #0d5de6;
  font-family: monospace;
}
```

(it's from the `<wpt-results>` component - I don't know a better way of sharing CSS between components :crying_cat_face:)
</blockquote></details>

Done.

---

*[components/wpt-runs.html, line 43 at r5](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPZ-r5-43:-KyHzcQto5gXTlL0YCJG:b-896fix) ([raw file](https://github.com/w3c/wptdashboard/blob/71920c92ce6f20feb4ecefbefc1056d98a622b11/components/wpt-runs.html#L43)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

These colors are a little confusing since green rectangles in the dashboard so far have only meant "tests passing." How about these:

```
.missing {
  background-color: #e3edfd;
}
.present {
  background-color: #337af3;
}
```

![screenshot from 2017-11-06 12 08 46](https://user-images.githubusercontent.com/916028/32461616-4c00161a-c2eb-11e7-92f7-51f08b101d92.png)

</blockquote></details>

Done.

---

*[components/wpt-runs.html, line 74 at r5](https://reviewable.io:443/reviews/w3c/wptdashboard/188#-Kxx50Q9Rz17kU4YZlPZ-r5-74:-KyHzdGXkZBWv9Yl0hQp:b-896fix) ([raw file](https://github.com/w3c/wptdashboard/blob/71920c92ce6f20feb4ecefbefc1056d98a622b11/components/wpt-runs.html#L74)):*
<details><summary><i>Previously, jeffcarp (Jeff Carpenter) wrote…</i></summary><blockquote>

This needs to be updated to be `sha=`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/wptdashboard/188)*
<!-- Sent from Reviewable.io -->

jeffcarp commented:
Excellent, thanks for this!

jeffcarp commented:
Deployed https://wpt.fyi/test-runs
